### PR TITLE
docs: Move general info about next.config.* to its index page

### PR DESCRIPTION
**Pull request by robertwbradford**
I was reading through the current https://nextjs.org/docs/app/api-reference/next-config-js/pageExtensions and noticed it started explaining different file formats for the `next.config.*` file. Is that information better suited for the https://nextjs.org/docs/app/api-reference/next-config-js index page?

There was also a little redundancy at the top.

### CODE REVIEW

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: 87a083f8539889dcbf7ce0b9334b7246c5b945fc

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: 87a083f8539889dcbf7ce0b9334b7246c5b945fc

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: 87a083f8539889dcbf7ce0b9334b7246c5b945fc

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

**Review by leerob — DISMISSED** [review]

File: docs/02-app/02-api-reference/05-next-config-js/index.mdx [hunk 1]
```diff
@@ -5,7 +5,7 @@ description: Learn how to configure your application with next.config.js.
 
 {/* The content of this doc is shared between the app and pages router. You can use the `<PagesOnly>Content</PagesOnly>` component to add content that is specific to the Pages Router. Any shared content should not be wrapped in a component. */}
 
-Next.js can be configured through a `next.config.js` file in the root of your project directory.
+Next.js can be configured through a `next.config.js` file in the root of your project directory (next to `package.json`).
```
Comment (leerob) [inline docs/02-app/02-api-reference/05-next-config-js/index.mdx:8]:
```suggestion
Next.js can be configured through a `next.config.js` file in the root of your project directory (i.e. by `package.json`).
```

**Review by leerob — APPROVED** [review]

**Pull request merged**